src/shaders/line.vertex.glsl

```diff
@@ -27,6 +39,9 @@ varying float v_gamma_scale;
 
 void main() {
     #pragma mapbox: initialize highp vec4 color
+    #pragma mapbox: initialize lowp float floorwidth
+    #pragma mapbox: initialize lowp vec4 dash_from
+    #pragma mapbox: initialize lowp vec4 dash_to
```

Comment:
Is there any risk of crossing the GL_MAX_VERTEX_ATTRIB for low-end devices for native? I recall that a common denominator for es 2 is 8 (safe to use). We are at 9 from this PR, I'm not sure if after this change all of these `#pragma`s may be used at once but there might be ways to consolidate them if that's a possibility. (Refer https://chromium.googlesource.com/angle/angle/+/HEAD/doc/ResourceLimits.md.)

Reply:
This is a really good point! I was thinking to wrap these too with #ifdefs but it had some issues, maybe this pragma is evaluated before them? I'll check how else could I wrap these.

Reply:
Seems very important to respect these limits, but just out of curiosity: since this PR consolidates but does not change the functionality, it should be possible to overcome this with judicious ifdef's or similar, right?

Reply:
I've looked into this a bit more and noticed that the shader for the dash line had also more than 8 attributes. It could be an issue on old devices, but since it was already implemented like that, this PR probably won't cause more issues.

If I understand it correctly `#pragma mapbox` will only produce attributes when the property is calculated. So with the attributes above, the cases when there could be more than 8 only happens when both gradient and dash are enabled, and at least 4 properties are calculated. Otherwise I think it should work fine.
For those cases maybe it would be good to implement a validation which was discussed here: https://github.com/mapbox/mapbox-gl-js/issues/4879

Reply:
@rreusser I think yes but also this PR enables the combined usage of dash and gradients, so technically it extends the scope of the shader a bit and might use more attributes when both of them are used at the same time.

Reply:
Maybe for now dash and gradient combined usage could be still disabled. On older devices it could be done with two passes to overcome the limitation, which could be implemented later.

Reply:
And another possible workaround: maybe I could pack `a_linesofar`, `a_uv_x`, and `a_split_index` into a vec3 and upload them as a single attribute. The downside is that they need an extra operation to unpack them, but this way we would have the same amount of attributes as we had before in `line_sdf.vertex.glsl`.
What do you think?